### ISSUE TITLE
Update icon examples with aria-labels

### ISSUE THREAD

**Issue by sowasred2012** [canonical/vanilla-framework#4032]
Follow up to this conversation: https://github.com/canonical-web-and-design/vanilla-framework/pull/4004#discussion_r726979773

In [icon accessibility docs](https://vanillaframework.io/docs/patterns/icons#accessibility), we suggest that the text meant for assistive technology should be between the `<i>` tags, and then we use CSS to place it offscreen for non-assistive tech users. We can achieve the same thing by moving that text to an `aria-label` attribute, and is how we should do it in examples and documentation.
